step03/app.py

```diff
@@ -0,0 +1,87 @@
+from flask import Flask, g, jsonify, send_from_directory, url_for
+import boto3
+import sqlite3
+from flask_sqlalchemy import SQLAlchemy
+from flask_migrate import Migrate
+
+app = Flask(__name__)
+app.extensions = {}
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///local.db'
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+
+db = SQLAlchemy(app)
+migrate = Migrate(app, db, compare_type=True)
+
+from models import User
+
+
+BUCKET_NAME = 'gg-thumbnail-project'
+# OUTPUT_FOLDER = 'thumbnails'
+UPLOAD_FOLDER = 'images'
+AWS_PROFILE = 'thumbnail-service'
+# AWS_PROFILE = 'freefood89'
+LOCAL_USER = {
+    'username': 'local',
+    'fullname': 'lolcats lolz',
+    'user_tier': 'noob'
+}
+
+
+session = boto3.Session(profile_name=AWS_PROFILE)
+s3 = session.client('s3')
+
+
+def init_db():
+    local_user = User.query.get(LOCAL_USER['username'])
+    new_local_user = User.from_dict(LOCAL_USER)
+    
+    if local_user:
+        db.session.merge(new_local_user)
+    else:
+        db.session.add(new_local_user)
+
+    db.session.commit()
+
+
+@app.cli.command('init_db')
+def init_local():
+    """Initializes the database for local testing"""
+    init_db()
+    print('Initialized the database.')
+
+
+@app.before_request
+def attach_current_user():
+    if app.config['DEBUG']:
+        g.current_user = LOCAL_USERNAME
```

Comment:
this was a typo in the template. all occurrences of `LOCAL_USERNAME` should have been `LOCAL_USER['username']`
 After fixing, make sure `localhost:5000/users/me` returns the local user

Reply:
Fixed all the `LOCAL_USERNAME` and the app is working. 

However, I did have some issues and was getting the following error. 

```
Traceback (most recent call last):
  File "app.py", line 15, in <module>
    from models import User
  File "/Users/j/FullStack_local/courses/apprentice/step03/models.py", line 1, in <module>
    from app import db
  File "/Users/j/FullStack_local/courses/apprentice/step03/app.py", line 15, in <module>
    from models import User
ImportError: cannot import name 'User' from 'models' (/Users/j/FullStack_local/courses/apprentice/step03/models.py)
```

Initially, I thought it shouldn't be an issue, but after searching it seems to be a circular dependency issue (still not completely sure). I moved the following `from models import User`  to inside of the `if  __name__ == __main__;` statement at the end and it works fine. 

I want to know if this is okay or not?

Reply:
How are you running the app? Are you using the flask cli?

`FLASK_DEBUG=1 FLASK_APP=application.py flask run`

Reply:
I was not using the flask cli, and running the `app.py` as a script.
Worked fine when I used the flask cli `flask fun`. 

Thanks.